### PR TITLE
[BUG] Mention limited preview options for Threshold and Event Correlation rules

### LINKED ISSUE #2238: [BUG] Mention limited preview options for Threshold and Event Correlation rules
## Description
When detection rules were switched to the new rule preview API in 8.1, the preview options for Threshold and Event Correlation rules were limited as follows:
- Threshold rules only had the option to preview rule results from the last hour
- Event Correlation rules only had the option to preview rule results from the last hour and the last day. 

The [rule preview docs](https://www.elastic.co/guide/en/security/8.3/rules-ui-create.html#preview-rules) for 8.1-8.3 don't mention this in step 2 --- can prob add a note mentioning this. 

## Related
- https://github.com/elastic/kibana/pull/116374
- https://github.com/elastic/enhancements/issues/16261

### PR DESCRIPTION
**Pull request by nastasha-solomon**
Fixes https://github.com/elastic/security-docs/issues/2238.

[Preview](https://security-docs_3683.docs-preview.app.elstc.co/guide/en/security/8.3/rules-ui-create.html#preview-rules) - Added note under step 2.

### CODE REVIEW

File: docs/detections/rules-ui-create.asciidoc [hunk 1]
```diff
@@ -266,9 +266,16 @@ NOTE: To preview rules, you need the `read` privilege for the `.preview.alerts-s
...
 . Write the rule query.
-+
 . Select a timeframe of data to preview query results -- *Last hour*, *Last day*, or *Last month* -- from the *Quick query preview* drop-down.
 +
+[NOTE] 
+=====
+Some rules have timeframe limitations:
+
+- *Threshold rules*: You can only preview query results from the last hour.
+- *Event Correlation rules*: You can only preview query results from the last hour and the last day.
```
Comment (natasha-moore-elastic) [inline docs/detections/rules-ui-create.asciidoc:276]: Nit:
```suggestion
- *Event correlation rules*: You can only preview query results from the last hour and the last day.
```

**Review by natasha-moore-elastic — APPROVED** [review]

**Review by dplumlee — APPROVED** [review]
These were the limitations we had in the specified releases, lgtm

**Review by yctercero — APPROVED** [review]

**Comment by nastasha-solomon** [comment]
@Mergifyio backport 8.2

**Comment by nastasha-solomon** [comment]
@Mergifyio backport 8.1

**Pull request merged**